Small output improvements to `vm list`

AliSoftware opened the pull request:
Very small improvements… but that will make our life easier when running some `hostmgr vm` commands locally

### `hostmgr vm start`

 - Print the `handle` and `IPv4Address` of the VM after it started

_(useful particularly when we don't provide an explicit `--handle` to `vm start` and wouldn't otherwise easily know which newly-created handle to subsequently run `vm details` on)_

### `hostmgr vm list`

- Remove unused `Architecture` column in `vm list` output
- Add `.running` VMState (corresponding to images located in `working-vm-images`)
- Add emojis for values in `Location` and `State` columns, to make the values more visually distinguishable
- Improve `Console.printTable` to support emojis + add column & header separators—making it more readable
- Make `vm list` now sort VMs by state first, then name

---

The result of `hostmgr vm list` now looks like this in a Terminal:

<img width="562" alt="image" src="https://github.com/Automattic/hostmgr/assets/216089/4449d118-0ab4-4b21-9fa4-7a05c6359484">

AliSoftware committed:
Show VM handle after successful `vm start`

AliSoftware committed:
Remove unused Architecture column in `vm list`

AliSoftware committed:
Add `.running` VMState + Emojis in `vm list`

AliSoftware committed:
Improve `Console.printTable`

 - To account for Emoji characters taking 2x the size of a monospaced regular character in a Terminal
 - To add support for a horizontal separator row after the header row

AliSoftware committed:
Make `vm list` sort VMs by state first, then name

AliSoftware committed:
Also print IP address of VM after `vm start`

jkmassel commented on Sources/hostmgr/commands/vm/VMStart.swift:
```diff
@@ -56,14 +57,17 @@ struct VMStartCommand: AsyncParsableCommand {
             if waitForever {
                 let timeout = Duration.seconds(1_000_000_000) // Doesn't crash like .greatestFiniteMagnitude
-                try await vmManager.waitForVMStartup(for: configuration, timeout: timeout)
+                ipAddress = try await vmManager.waitForVMStartup(for: configuration, timeout: timeout)
```
This is unlikely to work in a case where the machine needs to be set up (for instance, if we're making a new VM from scratch). I'd suggest double-checking that case, because `waitForVMStartup` relies on port 22 being open (which it wouldn't be on a brand-new VM)

AliSoftware replied:
Ah, good point, will double-check that case.

Just to confirm, if that's the case, that is not new to my change though, right? Since with `--wait-forever` on a new VM from scratch, it's always been the case that we'd never return from `waitForVMStartup` anyway (even before my PR)… so I think that behavior is expected with `--wait-forever` anyway?

AliSoftware replied:
I've [updated the instructions for when creating a new OS Template](https://github.com/Automattic/buildkite-ci/pull/380/commits/835b7d8807f90af8c8ab6a3454616e905fe1e70b) as part of [this PR](https://github.com/Automattic/buildkite-ci/pull/380), and confirmed by testing that, since the expected behavior is to call `vm create` first, then `vm start --persistent --wait-forever` in those cases, then it's expected that the `waitForVMStartup` will never return (and `vm start` will never finish and return prompt) in that particular case.

---

We _could_ potentially print the IP address _before_ waiting for SSH—by calling `try await ipAddress(forVmWithName: self.handle)` and printing the result _before_ the call to `waitForVMStartup`—so that the IP is shown even if we're configuring a VM from scratch that doesn't have SSH running yet. But:
 - That would have us call `ipAddress(forVmWithName:)` twice; or alternatively extract that call out of `waitForVMStartup`, but I don't like the semantics of separating those
 - The main goal of having the IP address printed is so we can SSH into the VM. While when we configure a brand new OS template from scratch, the only way to view the VM is via the "View" button of `hostmgr-helper` anyway. Printing the IP address in the console wouldn't help much in that particular case.

So instead I decided to update the wording of the log to `VM SSH IP Address: \(ipAddress)` to make it clear that this also implies that SSH is ready within the VM.

AliSoftware committed:
Show VM handle _before_ waiting for boot to finish

To be able to see it even on VMs created from scratch (like new macOS images), on which SSH might not yet be configured and for which we'd use `--wait-forever`, as `waitForVMStartup` would not return until SSH is configured and running in the VM

AliSoftware committed:
Print both Total and Allocated size in `vm list`

AliSoftware commented:
@jkmassel I ended up [fixing the `vm list` file size inconsistency with Finder discussed during our 1:1 after all](https://github.com/Automattic/hostmgr/pull/88/commits/30b9bda92fde902ce2fb0a041096a88deea9a470). We're now printing both Allocated file size (the one that was already printed before, which might account for APFS compression, I guess?) and Total file size (same value as reported by the Finder, and also matches the size used during `vm create --disk-size`)

<details><summary> 🖼️ Look of the new output</summary>
<img width="715" alt="image" src="https://github.com/Automattic/hostmgr/assets/216089/320f856a-1779-483c-86a4-7f9141d85cf4">
</details>

jkmassel commented:
To avoid confusion, I'd be fine to just use the total size for everything – is there a good reason not to?

AliSoftware committed:
Only print total size in `vm list`

AliSoftware commented:
Yeah, basically I added both values while debugging and trying to find the right `URLResourceKey` to match Finder's, and then figured I'd keep them as I already had the code… but you're probably right that in the end it showing both might be more confusing than helpful.

Fixed in https://github.com/Automattic/hostmgr/pull/88/commits/360cda341b59e4838da2680caad1f3eb80f8743f

AliSoftware committed:
Fix parameter name consistency

AliSoftware committed:
DRY `waitForVMStartup` call

AliSoftware committed:
Don't insert header separator if no columnTitles

AliSoftware committed:
Fix SwiftLint line length violations

jkmassel commented on Sources/libhostmgr/Platforms/LocalVMImage.swift:
```diff
@@ -52,7 +56,8 @@ public struct LocalVMImage: Equatable {
+public enum VMImageState: String, CaseIterable {
+    case packaged = "📦 Packaged"
+    case ready = "💾 Ready"
+    case running = "▶️ Running"
```
```suggestion
    case running = "🟢 Running"
```

Nit: The icon you specified renders ASCII-like in the terminal, which causes weird spacing issues in the table.

Not a blocker.

AliSoftware committed:
Adjust emoji for Running state

Co-authored-by: Jeremy Massel <1123407+jkmassel@users.noreply.github.com>
